controllers/bookController.js

```diff
@@ -12,14 +17,46 @@ exports.addBook = catchAsync(async (req, res, next) => {
   });
 });
 exports.getAllBooks = catchAsync(async (req, res, next) => {
-  const books = await Book.find();
+  const queryObj = { ...req.query };
+  const excludedFields = ["page", "sort", "limit"];
+
+  // Exclude fields that are not for filtering
+  excludedFields.forEach((el) => delete queryObj[el]);
+
+  // Advanced filtering
+  let queryStr = JSON.stringify(queryObj);
```

Comment:
You are giving the client to search with any attributes from your table. Why is that? Have you weighted the performance impact?

Reply:
i have few properties in my model for this but i will limit to just ISBN and genre.